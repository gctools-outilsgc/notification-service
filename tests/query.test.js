const query = require("../src/resolvers/Query");
const { getPrismaTestInstance } = require("./init/prismaTestInstance");

const parent = {};
const ctx = {
    prisma: getPrismaTestInstance()
};

test("Query all notifications", async() => {
  const info = "{id, gcID, appID, actionLevel, actionLink, email{ from, to, subject, body, status, html, sendError}, online{ titleEn, titleFr, descriptionEn, descriptionFr, viewed}, whoDunIt{ gcID, teamID, organizationID}}";
  
  expect(
    await query.notifications(parent, {}, ctx, info),
  ).toMatchSnapshot();

});

