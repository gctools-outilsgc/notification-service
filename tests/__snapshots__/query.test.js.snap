// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Query all notifications 1`] = `
Array [
  Object {
    "actionLevel": "Featured",
    "actionLink": null,
    "appID": "958742",
    "email": Object {
      "body": "Testing default email body",
      "from": "fromDefault@test.ca",
      "html": false,
      "sendError": null,
      "status": "Sent",
      "subject": "Test default subject",
      "to": "toDefault@test.ca",
    },
    "gcID": "234235fdg856bdgf",
    "online": Object {
      "descriptionEn": "Default Test online description English",
      "descriptionFr": "Test description par defaut en ligne en français",
      "titleEn": "Default test title English",
      "titleFr": "Test titre par defaut en français",
      "viewed": false,
    },
    "whoDunIt": Object {
      "gcID": "1a25sd487gd9",
      "organizationID": "325849",
      "teamID": "12698",
    },
  },
]
`;
